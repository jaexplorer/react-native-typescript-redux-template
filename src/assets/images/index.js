import Plus from './plus.png';
import Remove from './remove.png';

export { Plus, Remove };
