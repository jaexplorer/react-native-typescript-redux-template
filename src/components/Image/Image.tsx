import React from 'react';
import { Image as RNImage, ImageProps } from 'react-native';

const Image = (props: ImageProps): React.ReactElement<ImageProps> => {
  return <RNImage {...props} />;
};

export default Image;
