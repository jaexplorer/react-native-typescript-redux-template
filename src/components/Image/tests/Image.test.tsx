import React from 'react';
import { render } from '@testing-library/react-native';
import Image from '../Image';
import { Plus } from '../../../assets/images';

test('Does component render test', () => {
  const { baseElement } = render(<Image source={Plus} />);
  expect(baseElement).toMatchSnapshot();
});
