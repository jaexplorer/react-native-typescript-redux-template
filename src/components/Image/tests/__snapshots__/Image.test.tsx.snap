// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Does component render test 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/images/plus.png",
        }
      }
    />
  </View>
</View>
`;
